logstash-core/src/main/java/org/logstash/launchers/JvmOptionsParser.java

```diff
@@ -0,0 +1,297 @@
+
+        private static final long serialVersionUID = 2446165130736962758L;
+
+        private final Path jvmOptionsFile;
+
+        Path jvmOptionsFile() {
+            return jvmOptionsFile;
+        }
+
+        private final SortedMap<Integer, String> invalidLines;
+
+        SortedMap<Integer, String> invalidLines() {
+            return invalidLines;
+        }
+
+        JvmOptionsFileParserException(final Path jvmOptionsFile, final SortedMap<Integer, String> invalidLines) {
+            this.jvmOptionsFile = jvmOptionsFile;
+            this.invalidLines = invalidLines;
+        }
+
+    }
+
+    private final String logstashHome;
+
+    JvmOptionsParser(String logstashHome) {
+        this.logstashHome = logstashHome;
+    }
+
+    /**
+     * The main entry point. The exit code is 0 if the JVM options were successfully parsed, otherwise the exit code is 1. If an improperly
+     * formatted line is discovered, the line is output to standard error.
+     *
+     * @param args the args to the program which should consist of a single option, the path to LOGSTASH_HOME
+     */
+    public static void main(final String[] args) throws InterruptedException, IOException {
+        if (args.length != 1) {
+            throw new IllegalArgumentException(
+                    "Expected one arguments specifying path to LOGSTASH_HOME, but was " + Arrays.toString(args)
+            );
+        }
+
+        final JvmOptionsParser parser = new JvmOptionsParser(args[0]);
+        try {
+            parser.parse();
+        } catch (JvmOptionsFileParserException pex) {
+            System.err.printf(Locale.ROOT,
+                    "encountered [%d] error%s parsing [%s]",
+                    pex.invalidLines().size(),
+                    pex.invalidLines().size() == 1 ? "" : "s",
+                    pex.jvmOptionsFile());
+            int errorCounter = 0;
+            for (final Map.Entry<Integer, String> entry : pex.invalidLines().entrySet()) {
+                errorCounter++;
+                System.err.printf(Locale.ROOT,
+                        "[%d]: encountered improperly formatted JVM option in [%s] on line number [%d]: [%s]",
+                        errorCounter,
+                        pex.jvmOptionsFile(),
+                        entry.getKey(),
+                        entry.getValue());
+            }
+        } catch (IOException ex) {
+            System.err.println("Error accessing jvm.options file");
+            System.exit(1);
+        }
+    }
+
+    private void parse() throws IOException, JvmOptionsFileParserException {
+        final String jvmOpts = System.getenv("LS_JVM_OPTS");
+        // Set the initial JVM options from the jvm.options file. Look in
+        // /etc/logstash first, and break if that file is found readable there.
+        Optional<Path> jvmOptions;
+        if (jvmOpts == null || jvmOpts.isEmpty()) {
+            jvmOptions = Arrays.stream(new Path[] { Paths.get("/etc/logstash/jvm.options"),
```

Comment:
I'm not thrilled with bringing this POSIX-specific behaviour into the initialization for windows executables. I would prefer that this class required a valid path to a readable file (perhaps as an argument, not via an environment variable).

Reply:
I followed the nice suggestion to avoid reading sys environment variable `LS_JVM_OPTS` but bring it as argument.
About the POSIX-specific behaviour under windows it shouldn't be a problem, since `/etc/logstash/jvm.option` is not an existing path on Windows, so that path is skipped and check for existence of `$LS_HOME\config\jvm.options`, which is already the previous behavior in batch file